libs/labelbox/src/labelbox/schema/api_key.py

```diff
@@ -0,0 +1,407 @@
+
+        # Check if the user exists in the organization
+        user_id = ApiKey._get_user(client, user_email)
+        if not user_id:
+            raise ValueError(
+                f"User with email '{user_email}' does not exist in the organization"
+            )
+
+        role_name = role.name if hasattr(role, "name") else role
+        if not role_name or not isinstance(role_name, str):
+            raise ValueError("role must be a Role object or a valid role name")
+
+        allowed_roles = ApiKey._get_available_api_key_roles(client)
+        if role_name not in allowed_roles:
+            raise ValueError(
+                f"Invalid role specified. Allowed roles are: {allowed_roles}"
+            )
+
+        validity_seconds = 0
+        if validity > 0:
+            if not isinstance(time_unit, TimeUnit):
+                raise ValueError(
+                    "time_unit must be a valid TimeUnit enum value"
+                )
+
+            validity_seconds = validity * time_unit.value
+
+            if validity_seconds < TimeUnit.MINUTE.value:
+                raise ValueError("Minimum validity period is 1 minute")
+
+            max_seconds = 25 * TimeUnit.WEEK.value
+            if validity_seconds > max_seconds:
+                raise ValueError(
+                    "Maximum validity period is 6 months (or 25 weeks)"
+                )
+        else:
+            raise ValueError("validity must be a positive integer")
+
+        query_str = """
+         mutation CreateUserApiKeyPyApi($name: String!, $userEmail: String!, $role: String, $validitySeconds: Int) {
+             createApiKey(
+                 data: { name: $name, targetUserEmailId: $userEmail, role: $role, validitySeconds: $validitySeconds }
+             ) {
+                 id  
+                 jwt
+             }
+         }
+         """
+
+        params = {
+            "name": name,
+            "userEmail": user_email,
+            "role": role_name,
+            "validitySeconds": validity_seconds,
+        }
+
+        try:
+            result = client.execute(query_str, params)
+            api_key_result = result.get("createApiKey")
+
+            if not api_key_result:
+                raise LabelboxError(
+                    "Failed to create API key. No data returned from the server."
+                )
+
+            return api_key_result
+
+        except Exception as e:
+            if (
+                "permission" in str(e).lower()
+                or "unauthorized" in str(e).lower()
+            ):
+                raise LabelboxError(
+                    f"Permission denied: You don't have sufficient permissions to create API keys. Original error: {str(e)}"
+                )
+            else:
+                error_message = f"Failed to create API key: {str(e)}"
+                logger.error(error_message)
+                raise LabelboxError(error_message) from e
```

Comment:
here again I think most of that already handled by `.execute()` method and backend. It performs lots of extra checks and throws errors. So i think it is fine to just call `client.execute()` and return necessary data. But obviously should check before though 😬

Reply:
The exception handling was simplified.